File: optional/typescript-workflow-runner/src/application/jobs/system/hello-world.ts
#!/usr/bin/env node
/**
 * Hello World Workflow Step
 *
 * A simple workflow step example that can be executed via cron.
 * Works on both WSL (Linux) and Windows (Git Bash).
 *
 * Usage:
 *   npm run hello-world
 *   # or
 *   tsx src/jobs/system/hello-world.ts
 *
 * @module jobs/system/hello-world
 */

import "../../../infra/config/yaml.js";
import { createJob } from "../../job-factory.js";
import { registerJob } from "../../../registry/job-registry.js";
import { runIfMain } from "../../../runner/workflow-runner.js";
import { sleep } from "../../../shared/utils.js";

/**
 * Hello World job definition
 */
export const helloWorldJob = createJob(
  {
    name: "hello-world",
    description: "A simple hello world workflow step example",
  },
  async (_ctx, logger) => {
    // Simulate some work
    await sleep(100);

    const message = `Hello, World! Current time: ${new Date().toISOString()}`;
    logger.info({ message }, "Job executed successfully");

    return {
      success: true,
      message,
    };
  }
);

// Register with the job registry
registerJob(helloWorldJob);

// Export handler for direct execution
export const main = helloWorldJob.handler;

// Execute when run directly (CLI), skip when imported (Lambda/test)
runIfMain(main, import.meta.url);

File: starters/rust/batch/src/application/jobs/hello_world.rs
//! Hello World Batch Job
//!
//! A simple batch job example that can be executed via cron.
//!
//! Usage:
//!   cargo run -- --job hello-world

use std::sync::Arc;

use tracing::info;

use crate::application::job_factory::{create_job, JobDefinition, JobOptions};
use crate::application::job_types::JobResult;
use crate::di::AppContext;
use crate::shared::sleep_ms;

/// Create the hello-world job
pub fn hello_world_job() -> JobDefinition {
    create_job(
        JobOptions::new("hello-world", "A simple hello world batch job example"),
        |_ctx, app_ctx: Arc<AppContext>| async move {
            // Simulate some work
            sleep_ms(100).await;

            let message = format!(
                "Hello, World! Current time: {}",
                app_ctx.clock.to_iso_string()
            );

            info!(message = %message, "Job executed successfully");

            Ok(JobResult::success(message))
        },
    )
}
